sympy/parsing/c/c_parser.py

```diff
@@ -330,25 +359,30 @@ def transform_var_decl(self, node):
 
             except StopIteration:
 
-                if (node.type.kind == cin.TypeKind.INT):
-                    type = IntBaseType(String('integer'))
-                    value = Integer(0)
-                elif (node.type.kind == cin.TypeKind.FLOAT):
-                    type = FloatBaseType(String('real'))
-                    value = Float(0.0)
-                elif (node.type.kind == cin.TypeKind.BOOL):
-                    type = Type(String('bool'))
-                    value = false
+                if node.type.kind in self._data_types["int"]:
+                    type = self._data_types["int"][node.type.kind]
+                elif node.type.kind in self._data_types["float"]:
+                    type = self._data_types["float"][node.type.kind]
+                elif node.type.kind in self._data_types["bool"]:
+                    type = self._data_types["bool"][node.type.kind]
                 else:
                     raise NotImplementedError("Only bool, int "
-                            "and float are supported")
+                        "and float are supported")
+                value = None
 
-            return Variable(
-                node.spelling
-            ).as_Declaration(
-                type = type,
-                value = value
-            )
+            if value == None:
+                return Variable(
+                    node.spelling
+                ).as_Declaration(
+                    type = type
+                )
+            else:
+                return Variable(
+                    node.spelling
+                ).as_Declaration(
+                    type = type,
+                    value = value
+                )
```

Comment:
Both of these return the same thing. Value defaults to NoneToken anyway when not provided explicitly.

Reply:
Ohh, great one!